src/app/records/universal-ornaments/universal-ornaments.tsx

```diff
@@ -125,29 +126,23 @@ function identifyPlugSets(defs: D2ManifestDefinitions): { [classType: number]: n
  * using presentation node / collectible defs.
  */
 export const buildSets = memoizeOne((defs: D2ManifestDefinitions): OrnamentsData => {
-  const collectPresentationNodes = (
-    nodeHash: number,
-    list: DestinyPresentationNodeDefinition[]
-  ) => {
-    const def = defs.PresentationNode.get(nodeHash);
-    if (def && !def.redacted) {
-      if (def.children.collectibles.length) {
-        list.push(def);
-      }
-      for (const childNode of def.children.presentationNodes) {
-        collectPresentationNodes(childNode.presentationNodeHash, list);
-      }
-    }
-    return list;
-  };
-
   const plugSetHashes = identifyPlugSets(defs);
+  const collectibleFinder = createCollectibleFinder(defs);
   const data: OrnamentsData = {
     [DestinyClass.Titan]: { classType: DestinyClass.Titan, sets: {}, name: '', icon: '' },
     [DestinyClass.Hunter]: { classType: DestinyClass.Hunter, sets: {}, name: '', icon: '' },
     [DestinyClass.Warlock]: { classType: DestinyClass.Warlock, sets: {}, name: '', icon: '' },
   };
 
+  const findCollectibleArmorParentNode = (collectibleHash: number | undefined) => {
+    if (collectibleHash) {
+      const collectible = defs.Collectible.get(collectibleHash);
+      return collectible.parentNodeHashes
+        ?.map((nodeHash) => defs.PresentationNode.get(nodeHash))
```

Comment:
Out of paranoia, probably handle the case where either the collectible def or presentation node def are missing.